pkg/object/httpserver/mux.go

```diff
@@ -352,6 +353,14 @@ func (m *mux) reloadRules(superSpec *supervisor.Spec, muxMapper protocol.MuxMapp
 }
 
 func (m *mux) ServeHTTP(stdw http.ResponseWriter, stdr *http.Request) {
+	// HTTP-01 challenges requires HTTP server to listen on port 80, but we don't
+	// know which HTTP server listen on this port (consider there's an nginx sitting
+	// in front of Easegress), so all HTTP servers need to handle HTTP-01 challenges.
+	if strings.HasPrefix(stdr.URL.Path, "/.well-known/acme-challenge/") {
```

Comment:
Actually, we can get the port from `rules.spec.Port`.

Reply:
We cannot use this configuration here. The port needs to be accessible from the CA server,  so even `rules.spec.Port` is 80, it does not mean the CA server could send a request to this HTTP server, because other reverse proxies like Nginx could get the request and forward it to another HTTP server who does not listen on port 80.

Reply:
Oh we discussed this topic before, sure.